src/deps/boringssl.translated.zig

```diff
@@ -18851,6 +18877,117 @@ pub const struct_bio_st = extern struct {
     }
 };
 
+/// An X.509 Certificate.
+///
+/// > NOTE(@DonIsaac) I've started porting translated extern methods into this
+/// > struct, but not all have been moved. When you need an API not yet in here,
+/// > please move it.
+///
+/// ## References
+/// - [RFC 5280](https://datatracker.ietf.org/doc/html/rfc5280)
+/// - [BoringSSL - `x509.h` docs](https://commondatastorage.googleapis.com/chromium-boringssl-docs/x509.h.html)
+pub const X509 = opaque {
+    /// Parse an X.509 certificate from a PEM.
+    ///
+    /// Returns `null` if allocation failed, or if `pem_buffer` does not contain
+    /// a syntactically valid PEM.
+    ///
+    /// TODO(@DonIsaac): use ERR_get_error to construct an error variant.
+    pub fn initPEM(pem_buffer: *BIO) ?*X509 {
+        return PEM_read_bio_X509_AUX(pem_buffer, null, noPasswordCallback, null);
+    }
+
+    /// Parse an X.509 certificate from an ASN.1 DER encoded buffer.
+    ///
+    /// Returns `null` if allocation failed, or if `der_buffer` does not contain
+    /// a syntactically valid DER.
+    ///
+    /// TODO(@DonIsaac): use ERR_get_error to construct an error variant.
+    pub fn initDER(der_buffer: *BIO) ?*X509 {
+        return d2i_X509_bio(der_buffer, null);
```

Comment:
Does this clone the data in the bio?

Reply:
BIO is only readed here and X509 need to be `X509_free` when not needed anymore